diff --git a/src/fcch_detector.cc b/src/fcch_detector.cc
index 4c348cf..5632b1b 100644
--- a/src/fcch_detector.cc
+++ b/src/fcch_detector.cc
@@ -315,8 +315,7 @@ float fcch_detector::freq_detect(const complex *s, const unsigned int s_len, flo
 	fftw_execute(m_plan);
 
 	for(i = 0; i < FFT_SIZE; i++) {
-		fft[i].real() = m_out[i][0];
-		fft[i].imag() = m_out[i][1];
+		fft[i] = complex(m_out[i][0], m_out[i][1]);
 	}
 
 	max_i = peak_detect(fft, FFT_SIZE, &peak, &avg_power);
